Random crash in DocTab

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

DocTab randomly crashing when selecting an Item

### How to reproduce

I have no idea why sometimes it happens, and sometimes no...

### Screenshots

```
panic: runtime error: index out of range [1] with length 1

goroutine 38 [running]:
fyne.io/fyne/v2/container.(*docTabsRenderer).scrollToSelected(0xc0004e01b0?)
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/container/doctabs.go:362 +0x189
fyne.io/fyne/v2/container.(*docTabsRenderer).Refresh(0xc0004e01b0)
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/container/doctabs.go:267 +0x65 
fyne.io/fyne/v2/widget.(*BaseWidget).Refresh(0x7ff7f51bbf40?)
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/widget/widget.go:139 +0x2e     
fyne.io/fyne/v2/container.(*DocTabs).Select(0xc0001160a0, 0xc001ee48e0?)
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/container/doctabs.go:144 +0x37
main.(*MainWindow).TabAdd(0x7ff7f5743520, {0x7ff7f51bae60, 0xc0020cf880})
        C:/Users/Matwachich/Documents/Programmation/KissEMR/KissEMR/main_window.go:154 +0x147
main.(*tabHomeSearch).openOrCreatePatient(0xc0005044e0?, {0x93, 0xac, 0xb1, 0x43, 0x6b, 0x59, 0x45, 0xed, 0xb1, ...})
        C:/Users/Matwachich/Documents/Programmation/KissEMR/KissEMR/tab_home.go:237 +0x88
main.newTabHomeSearchListItem.func1()
        C:/Users/Matwachich/Documents/Programmation/KissEMR/KissEMR/tab_home.go:289 +0x5a
main.(*tabHomeSearchListItem).Tapped(0xc003348780, 0xc000448901?)
        C:/Users/Matwachich/Documents/Programmation/KissEMR/KissEMR/tab_home.go:334 +0x42
fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClickedHandleTapDoubleTap.func1()
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/internal/driver/glfw/window.go:634 +0x26
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/internal/driver/common/window.go:35 +0x3e
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1
        C:/Users/Matwachich/go/pkg/mod/fyne.io/fyne/v2@v2.3.4-rc1.0.20230516195251-56dd0f3e0f32/internal/driver/glfw/window.go:946 +0x136
exit status 2
```

### Example code

I have a complexe code base, I could not reproduce the bug with simple code.

### Fyne version

develop v2.3.4-rc1.0.20230516195251-56dd0f3e0f32

### Go compiler version

1.20

### Operating system and version

Windows 10

### Additional Information

_No response_

--- a/container/doctabs.go
+++ b/container/doctabs.go
@@ -363,6 +363,13 @@ func (r *docTabsRenderer) buildTabButtons(count int, buttons *fyne.Container) {
 
 func (r *docTabsRenderer) scrollToSelected() {
 	buttons := r.scroller.Content.(*fyne.Container)
+
+	// https://github.com/fyne-io/fyne/issues/3909
+	// very dirty temporary fix to this crash!
+	if r.docTabs.current < 0 || r.docTabs.current >= len(buttons.Objects) {
+		return
+	}
+
 	button := buttons.Objects[r.docTabs.current]
 	pos := button.Position()
 	size := button.Size()

